src/glmtools.jl

```diff
@@ -352,6 +374,10 @@ function devresid(::Binomial, y, μ)
 end
 devresid(::Gamma, y, μ) = -2 * (log(y / μ) - (y - μ) / μ)
 devresid(::InverseGaussian, y, μ) = abs2(y - μ) / (y * abs2(μ))
+function devresid(d::NegativeBinomial, y, μ)
+    θ = d.r
+    return μ == 0.0 ? NaN : 2 * (xlogy(y, y / μ) + xlogy(y + θ, (μ + θ)/(y + θ)))
```

Comment:
Should use `oftype`. I guess the most reliable solution is to compute the second operand anyway and store it in a variable, and use `oftype(v, NaN)`.

BTW, `μ == 0` is enough.